src/string/string-hashing.md

```diff
@@ -188,7 +188,7 @@ Implementation:
             long long cur_h = h [i + l-1];
             if (i) cur_h - = h [i-1];
             // Get all the hashes to the same degree
-            cur_h * = p_pow [ni-1];
```

Comment:
The original article has `n-i-1`, which looks correct, because `n-i` can equal `n` which will be outside of `p_pow` array.